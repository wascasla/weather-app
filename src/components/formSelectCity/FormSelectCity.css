.form-select-city{
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
}

.select-city {
  width: 70%;
  margin: auto;
}