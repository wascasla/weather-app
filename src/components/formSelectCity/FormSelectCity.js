import React, { useState } from "react";
import "./FormSelectCity.css";

const FormSelectCity = ({ getWeather }) => {
  const [listCities] = useState([
    {
      id: 1,
      name: "Miami",
      latitude: 25.774269,
      longitude: -80.193657,
    },
    {
      id: 2,
      name: "Porto Alegre",
      latitude: -30.03306,
      longitude: -51.23,
    },
    {
      id: 3,
      name: "Tilcara",
      latitude: -23.57757,
      longitude: -65.350899,
    },
    {
      id: 4,
      name: "Villa del Prado",
      latitude: 40.26915,
      longitude: -4.3046,
    },
    {
      id: 5,
      name: "New York City",
      latitude: 40.714272,
      longitude: -74.005966,
    },
  ]);

  const [citySelected, setCitySelected] = useState(0);

  const handleChange = (event) => {
    const result = listCities.find((element) => element.id === parseInt(event.target.value));
    setCitySelected(result);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    if (citySelected) {
      getWeather(citySelected);
    } else {
      alert("Debe seleccionar una ciudad!!!");
    }
  };

  return (
    <div className="select-city">
      <form className="form-select-city" onSubmit={handleSubmit}>
        <label>Elige una ciudad para consultar el clima:</label>
        <select name="city" value={citySelected?.id} defaultValue="none" onChange={handleChange}>
          <option value="none" disabled hidden>
            Seleccione una Ciudad
          </option>
          {listCities.map((city) => (
            <option key={city.id} value={city.id}>
              {city.name}{" "}
            </option>
          ))}
        </select>

        <input type="submit" value="Consultar" />
      </form>
    </div>
  );
};

export default FormSelectCity;
