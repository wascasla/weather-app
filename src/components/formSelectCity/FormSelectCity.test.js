import React from "react";
import { shallow } from "enzyme";
import FormSelectCity from "./FormSelectCity";

describe(" Pruebas en <FormSelectCity />", () => {
  const getWeather = jest.fn();
  let wrapper = shallow(<FormSelectCity getWeather={getWeather} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<FormSelectCity getWeather={getWeather} />);
  });

  test("Debe de mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("No debe de llamar a la funciÃ³n si no selecciona una ciudad", () => {
    window.alert = jest.fn();
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(getWeather).not.toHaveBeenCalled();
    expect(window.alert).toHaveBeenCalled();
    //expect(wrapper.find("alert").exists()).toBe(true);
  });
});
