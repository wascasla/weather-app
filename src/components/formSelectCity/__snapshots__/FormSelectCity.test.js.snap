// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Pruebas en <FormSelectCity /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="select-city"
>
  <form
    className="form-select-city"
    onSubmit={[Function]}
  >
    <label>
      Elige una ciudad para consultar el clima:
    </label>
    <select
      defaultValue="none"
      name="city"
      onChange={[Function]}
    >
      <option
        disabled={true}
        hidden={true}
        value="none"
      >
        Seleccione una Ciudad
      </option>
      <option
        key="1"
        value={1}
      >
        Miami
         
      </option>
      <option
        key="2"
        value={2}
      >
        Porto Alegre
         
      </option>
      <option
        key="3"
        value={3}
      >
        Tilcara
         
      </option>
      <option
        key="4"
        value={4}
      >
        Villa del Prado
         
      </option>
      <option
        key="5"
        value={5}
      >
        New York City
         
      </option>
    </select>
    <input
      type="submit"
      value="Consultar"
    />
  </form>
</div>
`;
